service/matching/fx.go

```diff
@@ -0,0 +1,144 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package matching
+
+import (
+	"context"
+
+	"go.uber.org/fx"
+	"google.golang.org/grpc"
+
+	"go.temporal.io/server/common"
+	"go.temporal.io/server/common/config"
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/metrics"
+	persistenceClient "go.temporal.io/server/common/persistence/client"
+	esclient "go.temporal.io/server/common/persistence/visibility/store/elasticsearch/client"
+	"go.temporal.io/server/common/resource"
+	"go.temporal.io/server/common/rpc/interceptor"
+	"go.temporal.io/server/service"
+	"go.temporal.io/server/service/history/configs"
+)
+
+var Module = fx.Options(
+	fx.Provide(ParamsExpandProvider), // BootstrapParams should be deprecated
+	fx.Provide(dynamicconfig.NewCollection),
+	fx.Provide(NewConfig),
+	fx.Provide(PersistenceMaxQpsProvider),
+	fx.Provide(ThrottledLoggerRpsFnProvider),
+	fx.Provide(TelemetryInterceptorProvider),
+	fx.Provide(RateLimitInterceptorProvider),
+	fx.Provide(service.GrpcServerOptionsProvider),
+	resource.Module,
```

Comment:
move Modules to the top of the list.

Reply:
I think it's reasonable to put dependencies in the order of supposed initialization. Makes it a bit easier to track. But I can definitely reorder.